### PR TITLE
Fix/shadowrootmode

### PR DESCRIPTION
**Pull request by spmonahan**
# Pull Request

## 📖 Description

Changes the `shadowroot` attribute on `<template>`s to `shadowrootmode` per the standardized declarative shadow DOM when server rendering components.

## 👩‍💻 Reviewer Notes

[web.dev](https://developer.chrome.com/docs/css-ui/declarative-shadow-dom) notes that pre-standardization, Chrome and Edge used the `shadowroot` attribute instead. This attribute doesn't work in any current version of the browsers I tested in so I've removed it entirely. This seems a reasonable choice to me since this package is still in beta and the versions of Chrome and Edge that support the non-standard attribute are quite old at this point, however, if it's necessary to support the non-standard attribute I'm happy to do that work but I will need guidance on how to test against older browser versions.

## 📑 Test Plan

I updated relevant tests in the repo and validated the before and after in Edge, Chrome, Safari and Firefox.

[This is my test repo](https://github.com/spmonahan/fast-ssr).

#### Edge
edge_shadowroot

#### Chrome
chrome_shadowroot

#### Firefox
firefox_shadowroot

#### Safari
safari_shadowroot

#### Edge
edge_shadowrootmode

#### Chrome
chrome_shadowrootmode

#### Firefox
firefox_shadowrootmode

#### Safari
safari_shadowrootmode

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: change/@microsoft-fast-ssr-b7eb9d42-4f98-4cae-842d-f8792c26a757.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: change 'shadowroot' to 'shadowrootmode' to match spec",
+  "packageName": "@microsoft/fast-ssr",
+  "email": "seanmonahan@microsoft.com",
+  "dependentChangeType": "patch"
```
Comment (scomea) [inline change/@microsoft-fast-ssr-b7eb9d42-4f98-4cae-842d-f8792c26a757.json:6]: "prerelease" here too

**Review by scomea — APPROVED** [review]
Seems straightforward

**Review by KingOfTac — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

File: change/@microsoft-fast-ssr-b7eb9d42-4f98-4cae-842d-f8792c26a757.json (hunk 1 above)
Comment (spmonahan) [inline change/@microsoft-fast-ssr-b7eb9d42-4f98-4cae-842d-f8792c26a757.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

Comment (spmonahan) [reply change/@microsoft-fast-ssr-b7eb9d42-4f98-4cae-842d-f8792c26a757.json]: done!

**Pull request merged**